webapi/src/main/scala/org/knora/webapi/responders/v2/SearchResponderV2.scala

```diff
@@ -325,6 +360,120 @@ final case class SearchResponderV2Live(
     gravsearchV2(query, rendering, user, limitToProject)
   }
 
+  /**
+   * Performs a Gravsearchquery to find StillImageRepresentations that link to the specified resource.
+   *
+   * @param resourceIri     the IRI of the resource to which StillImageRepresentations are to be found.
+   * @param offset          the offset to be used for paging.
+   * @param rendering       the schema of the response.
+   * @param user            the client making the request.
+   * @param limitToProject  the option to limit the search to the specific project.
+   * @return a [[ReadResourcesSequenceV2]] representing the resources that have been found.
+   */
+  def searchStillImageRepresentationsV2(
+    resourceIri: IRI,
+    offset: Int,
+    rendering: SchemaRendering,
+    user: User,
+    limitToProject: Option[ProjectIri],
+  ): Task[ReadResourcesSequenceV2] = {
+    val query: String =
+      s"""
+         |PREFIX knora-api: <http://api.knora.org/ontology/knora-api/simple/v2#>
+         |
+         |CONSTRUCT {
+         |?page knora-api:isMainResource true .
+         |
+         |?page knora-api:seqnum ?seqnum .
+         |
+         |?page knora-api:hasStillImageFile ?file .
+         |} WHERE {
+         |
+         |?page a knora-api:StillImageRepresentation .
+         |?page a knora-api:Resource .
+         |
+         |?page knora-api:isPartOf <$resourceIri> .
+         |knora-api:isPartOf knora-api:objectType knora-api:Resource .
+         |
+         |<$resourceIri> a knora-api:Resource .
+         |
+         |?page knora-api:seqnum ?seqnum .
+         |knora-api:seqnum knora-api:objectType xsd:integer .
+         |
+         |?seqnum a xsd:integer .
+         |
+         |?page knora-api:hasStillImageFile ?file .
+         |knora-api:hasStillImageFile knora-api:objectType knora-api:File .
+         |
+         |?file a knora-api:File .
+         |
+         |} ORDER BY ?seqnum
+         |OFFSET $offset
+         |""".stripMargin
+
+    gravsearchV2(query, rendering, user, limitToProject)
+  }
+
+  /**
+   * Performs a Gravsearchquery to find StillImageRepresentations count linked to the specified resource.
+   *
+   * @param resourceIri     the IRI of the resource to which StillImageRepresentations are to be found.
+   * @param user            the client making the request.
+   * @param limitToProject  the option to limit the search to the specific project.
+   * @return a [[ResourceCountV2]] representing the number of StillImageRepresentations that have been found.
+   */
+  def searchStillImageRepresentationsCountV2(
+    resourceIri: IRI,
+    user: User,
+    limitToProject: Option[ProjectIri],
+  ): Task[ResourceCountV2] = {
+    val query: String =
+      s"""
```

Comment:
Shouldn't this be the exact same query as in the search case?

If so, IMHO, we should not be duplicating it then.

Reply:
True, these queries are almost the same. I thought to inline it but then forgot. Thanks for the reminder. Done in [5e60efe](https://github.com/dasch-swiss/dsp-api/pull/3573/commits/5e60efee4ec418a5b295d7f003ca4b37165964a0).